### PR TITLE
Add createRepo public endpoint

### PR DESCRIPTION
**Pull request by riya-n**
- this should create a trunk repo for orgs that don't already have them
- calls the createRepo public endpoint: https://github.com/trunk-io/trunk/pull/14243
- tested upload on a test repo and seems to work fine
- TODO: wait for endpoint to be on prod, then merge this and update cli version on uploader

### CODE REVIEW

**Comment by gnalh** [comment]
Is this endpoint available on prod yet?

**Review by gnalh — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -268,7 +266,19 @@ async fn run_upload(
     }
 
     Retry::spawn(default_delay(), || {
-        trunk_analytics_cli::clients::put_bundle_to_s3(&upload.url, &bundle_time_file)
+        put_bundle_to_s3(&upload.url, &bundle_time_file)
+    })
+    .await?;
+
+    let remote_urls = vec![repo.repo_url.clone()];
```
Comment (zaycev) [inline src/main.rs:273]: Is this a correct order? Should we create repo first and then create upload?

File: src/types.rs [hunk 2]
```diff
@@ -56,6 +65,12 @@ pub struct QuarantineBulkTestStatus {
     pub quarantine_results: Vec<QuarantineResult>,
 }
 
+#[derive(Debug, Serialize, Clone, Deserialize)]
+pub struct TrunkRepo {
```
Comment (zaycev) [inline src/types.rs:69]: If we aren't using response object, maybe we can have empty schema for now.

File: src/main.rs (hunk 1 above)
Comment (gnalh) [reply src/main.rs:273]: The order shouldn't matter since the other endpoints don't depend on a repo existing.

File: src/types.rs (hunk 2 above)
Comment (riya-n) [reply src/types.rs:69]: Yeah, we aren't using it, but I think it's fine to leave this as a sanity check that the repoId exists in the response

**Pull request merged**